core/vm/contracts.go

```diff
@@ -1135,3 +1142,29 @@ func kZGToVersionedHash(kzg kzg4844.Commitment) common.Hash {
 
 	return h
 }
+
+// P256VERIFY (secp256r1 signature verification)
+// implemented as a native contract
+type p256Verify struct{}
+
+// RequiredGas returns the gas required to execute the precompiled contract
+func (c *p256Verify) RequiredGas(input []byte) uint64 {
+	return params.P256VerifyGas
+}
+
+// Run executes the precompiled contract, returning the output and the used gas
+func (c *p256Verify) Run(input []byte) ([]byte, error) {
+	// Required input length is 160 bytes
+	const p256VerifyInputLength = 160
+
+	// "input" is (hash, r, s, x, y), each 32 bytes
+	input = common.RightPadBytes(input, p256VerifyInputLength)
+
+	// Extract the hash, r, s, x, y from the input
+	hash := input[0:32]
+	r, s := new(big.Int).SetBytes(input[32:64]), new(big.Int).SetBytes(input[64:96])
+	x, y := new(big.Int).SetBytes(input[96:128]), new(big.Int).SetBytes(input[128:160])
+
+	// Verify the secp256r1 signature
+	return secp256r1.Verify(hash, r, s, x, y)
```

Comment:
This may be more a comment about the EIP, and less about this PR, but you should really consider the implications of returning an error here. When a precompile returns an error, the call will revert. Not sure if this is intended. The EIP draft certain does not speak of reverts, it just mentions that 

> If the given `x` and `y` coordinates are not valid, it returns null for the result with the given error.

This statement has no meaning in EVM, because it has no concept of returning a result and error at the same time.

Reply:
Thanks, these are valuable feedbacks. I see that there is a confusion in the narrative style between the cryptographic and code implementation. I will make required changes in the EIP text after resolving implementation in the other reply' part, asking related question there.

Reply:
Planning to simplify the related part in the EIP just like this:

> 
> - **Output data:** 32 bytes of result data and error
>    - If the signature verification process succeeds, it returns 1 in 32 bytes format.